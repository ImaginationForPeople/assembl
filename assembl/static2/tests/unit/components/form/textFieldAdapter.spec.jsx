import React from 'react';
import renderer from 'react-test-renderer';

import DummyForm from './dummyForm';
import TextFieldAdapter from '../../../../js/app/components/form/textFieldAdapter';

describe('TextFieldAdapter component', () => {
  const onChangeSpy = jest.fn();
  const onFocusSpy = jest.fn();

  it('should render a text field', () => {
    const props = {
      input: {
        name: 'fullname',
        onChange: onChangeSpy,
        onFocus: onFocusSpy,
        value: 'Amber Zieme'
      },
      label: 'Full name',
      meta: {
        error: '',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <TextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field without value', () => {
    const props = {
      input: {
        name: 'fullname',
        onChange: onChangeSpy,
        onFocus: onFocusSpy
      },
      label: 'Full name',
      meta: {
        error: '',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <TextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field untouched with error', () => {
    const props = {
      input: {
        name: 'fullname',
        onChange: onChangeSpy,
        onFocus: onFocusSpy,
        value: 'Amber Zieme'
      },
      label: 'Full name',
      meta: {
        error: 'This field is required',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <TextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field touched with error', () => {
    const props = {
      input: {
        name: 'fullname',
        onChange: onChangeSpy,
        onFocus: onFocusSpy,
        value: 'Amber Zieme'
      },
      label: 'Full name',
      meta: {
        error: 'This field is required',
        touched: true
      }
    };
    const component = renderer.create(
      <DummyForm>
        <TextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});