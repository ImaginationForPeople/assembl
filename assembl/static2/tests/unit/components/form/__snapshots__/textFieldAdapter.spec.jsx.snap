// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldAdapter component should render a text field 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="fullname"
    >
      Full name
    </label>
    <input
      className="form-control"
      id="fullname"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Full name"
      type={undefined}
      value="Amber Zieme"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`TextFieldAdapter component should render a text field touched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group has-error"
  >
    <label
      className="control-label"
      htmlFor="fullname"
    >
      Full name
    </label>
    <input
      className="form-control"
      id="fullname"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Full name"
      type={undefined}
      value="Amber Zieme"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`TextFieldAdapter component should render a text field untouched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="fullname"
    >
      Full name
    </label>
    <input
      className="form-control"
      id="fullname"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Full name"
      type={undefined}
      value="Amber Zieme"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`TextFieldAdapter component should render a text field without value 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="fullname"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Full name"
      type={undefined}
      value={undefined}
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
